subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/junitplatform/JUnitPlatformIntegrationTest.groovy

```diff
@@ -52,6 +78,84 @@ class JUnitPlatformIntegrationTest extends JUnitPlatformIntegrationSpec {
         succeeds('test')
     }
 
+    def 'can handle dynamically created test'() {
+        given:
+        buildFile << """
+        dependencies {
+            testImplementation 'org.junit.platform:junit-platform-runner:1.0.3'
+        }
+        """
+        createDynamicJupiterTest()
+
+        when:
```

Comment:
This when/then setup looking more complicated than need. Probably same way as JUnitPlatformLoggingIntegrationTest doing will allow you to achieve same behavior more ideomatically?